.github/workflows/ci.yml

```diff
@@ -387,6 +387,31 @@ jobs:
         sudo ./ci/setup_grpc.sh
         ./ci/do_ci.sh cmake.exporter.otprotocol.test
 
+  cmake_modern_protobuf_grpc_test:
+    name: CMake test (with modern protobuf and grpc)
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+    - name: setup
+      env:
+        PROTOBUF_VERSION: '23.2'
+        ABSEIL_CPP_VERSION: '20230125.3'
+        CXX_STANDARD: '17'
+      run: |
+        sudo ./ci/setup_cmake.sh
+        sudo ./ci/setup_ci_environment.sh
+        sudo -E ./ci/install_abseil.sh
+        sudo -E ./ci/install_protobuf.sh
+    - name: run otlp exporter tests
+      env:
+        WITH_ABSEIL: 'ON'
```

Comment:
Should we also have a test without WITH_ABSEIL to ensure that ABI compatible build works fine ?

Reply:
The latest versions of protobuf and gRPC require the new abseil-cpp, so we must enable it when using these versions.

Reply:
But `WITH_ABSEIL` cmake option will enable `HAVE_ABSEIL` macro, and so enforce to use `abseil::variant` at API surface. Do you mean with latest versions of gRPC and protobuf, we can no longer use `nostd::variant` at API surface ?

Reply:
Thanks for confirming. This looks concerning to me.

Reply:
Yes, it'a problem.Do you have any suggestion?

Reply:
Not for now. I would like to test the build with newer protobuf to fully understand the issue with our internal nostd::variant, and if we can do any fix in that class. Will try to test during this week. If not, good to discuss this in the next community meeting before deciding.
Thanks once again for providing all the details.

Reply:
Can we raise another issue to track this ABI compliance problem later? This problem existed before with abseil.

Reply:
Can we wait for Monday SIG meeting, also would like to test this once over weekend to have better understanding. Don't want to hold this PR any way as we need to support latest protobuf.

Reply:
Thanks.